extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -409,6 +409,19 @@ public void handle(RoutingContext event) {
                 && hotReplacementContext.getDevModeType() == DevModeType.REMOTE_SERVER_SIDE) {
             root = remoteSyncHandler = new RemoteSyncHandler(liveReloadConfig.password.get(), root, hotReplacementContext);
         }
+
+        if (launchMode == LaunchMode.DEVELOPMENT) {
+            //vert.x threads can still reference the old TCCL
+            ClassLoader newCl = Thread.currentThread().getContextClassLoader();
+            Handler<HttpServerRequest> old = root;
+            root = new Handler<HttpServerRequest>() {
+                @Override
+                public void handle(HttpServerRequest event) {
+                    Thread.currentThread().setContextClassLoader(newCl);
+                    old.handle(event);
```

Comment:
If `old.handle` was defined like this, it will still set its TCCL to the old TCCL it captured as the first thing it does right after you've just set the new TCCL, no?

Reply:
Good point.

This is not actually supposed to be part of this PR, I have been working on this on and off for a while and I must have accidentally included an unrelated change. This has actually already been fixes as part of https://github.com/quarkusio/quarkus/commit/6b6db29a8f9af2146a497a688f359b78cabccabe